plutarch-test/plutarch-base/Plutarch/ApiSpec.hs

```diff
@@ -46,6 +48,68 @@ spec = do
             $ \case
               PMinting c -> popaque c
               _ -> perror
+    describe "value" $ do
+      pgoldenSpec $ do
+        let pmint = PValue.singleton # pconstant "c0" # pconstant "sometoken" # 1
+            pmintOtherToken = PValue.singleton # pconstant "c0" # pconstant "othertoken" # 1
+            pmintOtherSymbol = PValue.singleton # pconstant "c2" # pconstant "sometoken" # 1
+        "singleton" @| pmint @-> \p ->
+          plift p @?= mint
+        "valueOf" @| PValue.valueOf # pmint # pconstant "c0" # pconstant "sometoken" @-> \p ->
+          plift p @?= 1
+        "unionWith" @\ do
+          "const" @| PValue.unionWith # plam const # pmint # pmint @-> \p ->
+            plift p @?= mint
+          "(+)" @| PValue.unionWith # plam (+) # pmint # pmint @-> \p ->
+            plift p @?= mint <> mint
+          "tokens" @| PValue.unionWith # plam (+) # pmint # pmintOtherToken @-> \p ->
+            plift p @?= mint <> mintOtherToken
+          "symbols" @| PValue.unionWith # plam (+) # pmint # pmintOtherSymbol @-> \p ->
+            plift p @?= mint <> mintOtherSymbol
+    describe "map" $ do
+      pgoldenSpec $ do
+        let pmap, pdmap, emptyMap, doubleMap, otherMap :: Term _ (AssocMap.PMap PByteString PInteger)
+            pmap = AssocMap.singleton # pconstant "key" # 42
+            pdmap = AssocMap.singletonData # pdata (pconstant "key") # pdata 42
+            emptyMap = AssocMap.empty
+            doubleMap = AssocMap.singleton # pconstant "key" # 84
+            otherMap = AssocMap.singleton # pconstant "newkey" # 6
+        "lookup" @| AssocMap.lookup # pconstant "key" # pmap #== pcon (PJust 42) @-> passert
```

Comment:
Ok, so the general advice when writing golden programs is to keep assertion checks out of the golden itself. So this should instead look like:

```haskell
"lookup" 
  @| AssocMap.lookup # pconstant "key" # pmap
  @-> \result -> passert $ result #== pcon (PJust 42)
```

The idea is to store the `lookup` function application _only_ in the golden file. Only the expression being tested is made a golden value. The comparing of its evaluated result happens outside, specifically in the `@->` assertion. 

(Likewise for other golden below).

Reply:
Basically typically wherever `@-> passertNot` or `@-> passert` is used.

Reply:
Done.